target/linux/x86/base-files/lib/preinit/01_sysinfo

```diff
@@ -8,7 +17,7 @@ do_sysinfo_x86() {
 
 	for file in product_name board_name; do
 		product="$(cat /sys/devices/virtual/dmi/id/$file 2>/dev/null)"
-		[ -n "$product" ] && break
+		[ -n "$product" -o \( "$vendor" = "Supermicro" -a "$product" = "Super Server" \) ] && break
```

Comment:
Why is this special casing needed, can you elaborate?

Reply:
Because the Supermicro "Super Server" family encompasses about 40+ different platforms all with different capabilities. Even the same basic motherboard can have a 2- to 24-core CPU on it, which means that no useful determination of capabilities, hardware, etc. can be made from this name.  Therefore, we might as well ignore this `product_name` when we see it and use the `board_name` which is much more descriptive.

I have, for instance, an `X10SDV-8C-TLN4F` which is an 8-core Xeon D-1541 motherboard with 2x i350 GBE ports, and 2x x552 10GBE ports.  I also have an `X10SDV-TP8F` which is a 4-core Xeon D-1518 with 2x i210 GBE ports, 4x i350 GBE ports, and 2x x552 10GBE ports.

Both have the same `product_name` which makes it useless.

Reply:
If you want, I could also structure this as:

```
case "$vendor:$product" in
*:)
        continue ;;
Supermicro:"Super Server")
        continue ;;
esac
break
```

if that's more legible.

Reply:
Yes, please refactor into a case switch.